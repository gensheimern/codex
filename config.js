module.exports = {
	DB_HOST: process.env.DB_HOST || '159.65.115.221',
	DB_USER: process.env.DB_USER || 'root',
	DB_PASS: process.env.DB_PASS || 'teamcodex',
	DB_NAME: process.env.DB_NAME || 'lunch_planner',
	JWT_SECRET: process.env.JWT_SECRET || 'secret',
	MAIL_HOST: process.env.MAIL_HOST || '159.65.115.221',
	MAIL_USER: process.env.MAIL_USER || 'support@codex-team.de',
	MAIL_PASS: process.env.MAIL_PASS || 'teamcodex123',
};
