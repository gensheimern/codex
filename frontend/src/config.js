const config = {
	debug: process.env.REACT_APP_DEBUG === 'TRUE',
	apiPath: process.env.REACT_APP_API_PATH,
	wsPath: process.env.REACT_APP_WS_PATH,
	basePath: process.env.REACT_APP_BASE_PATH,
}

export default config;