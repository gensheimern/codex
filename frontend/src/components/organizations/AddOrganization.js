import React from 'react';
import TextField from 'material-ui/TextField';
import RaisedButton from 'material-ui/RaisedButton';
import config from '../../config';

/**
 * This component show a dialog to add an organization.
 */
export default class AddOrganization extends React.Component {
	constructor(props) {
		super(props);
	
		this.state = {
			name: '',
			description: '',
			password: '',
			retypePassword: '',
			createdError: false,
			clicked: false,
		};

		this.createOrganization = this.createOrganization.bind(this);
	}

	createOrganization() {
		this.setState({
			createdError: false,
		});

		fetch(`${config.apiPath }/organization`, {
            method: 'POST',
            headers: {
				'Content-Type': 'application/json',
				'X-Access-Token': localStorage.getItem('apiToken'),
			},
			body: JSON.stringify({
				name: this.state.name,
				description: this.state.description,
				password: this.state.password,
			}),
		})
		.then((res) => {
            if(!res.ok) {
                throw new Error("Response not ok.");
            } else if(res.status !== 201) {
                throw new Error("An error occured.");
            }
            return res.json();
		})
		.then((res) => {
            this.setState({
				createdError: false,
			});

			this.joinOrganization(res.organizationId, this.state.password)
				.then(this.props.reload).catch();

			this.props.back();
        }).catch((error) => {
			this.setState({
				createdError: true,
			});
        });
	}

	joinOrganization(organizationId, password) {
		return fetch(`${config.apiPath }/user/me/organizations/${organizationId}`, {
            method: 'POST',
            headers: {
				'Content-Type': 'application/json',
				'X-Access-Token': localStorage.getItem('apiToken'),
			},
			body: JSON.stringify({
				password,
			}),
		});
	}

	handleChange = name => e => {
		this.setState({
			[name]: e.target.value
		});
	}

	differentPasswords = () => {
		return this.state.password !== this.state.retypePassword;
	}

	validParams = () => {
		return this.state.name
			&& this.state.password
			&& this.state.retypePassword
			&& this.state.password === this.state.retypePassword;
	}

	render() {
		const errorMessage = (
			<p style={{
				color: 'red',
			}}>Invalid values.</p>
		);

		return (
			<React.Fragment>
				<h4>Create Organization:</h4>
				<TextField
					id="AOname"
					type="text"
					floatingLabelText="Name"
					value={this.state.name}
					onChange={this.handleChange('name')}
					style={{
						width: '100%',
					}}
				/>
				<br/>
				<TextField
					id="AOdescription"
					type="text"
					floatingLabelText="Description"
					value={this.state.description}
					onChange={this.handleChange('description')}
					style={{
						width: '100%',
					}}
				/>
				<br/>
				<TextField
					id="AOpassword"
					type="password"
					floatingLabelText="Password"
					value={this.state.password}
					onChange={this.handleChange('password')}
					style={{
						width: '100%',
					}}
				/>
				<br/>
				<TextField
					id="AOretypePassord"
					type="password"
					floatingLabelText="Retype password"
					value={this.state.retypePassword}
					onChange={this.handleChange('retypePassword')}
					errorText={this.differentPasswords() ? 'Passwords not matching' : null}
					style={{
						width: '100%',
					}}
				/>
				<br/>

				<RaisedButton
					id="AOcreate"
					label="Create"
					primary={true}
					disabled={!this.validParams()}
					onClick={() => {
						if (this.state.clicked) return;
						this.setState({
							clicked: true,
						});
						this.createOrganization();
					}}
					style={{
						float: 'right',
					}}
				/>
				{this.state.createdError ? errorMessage : null}
			</React.Fragment>
		);
	}
}
