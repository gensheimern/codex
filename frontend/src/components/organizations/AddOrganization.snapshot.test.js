/**
 * @author:Stella Neser
 *  Komponente ist ein Snapshottest
 */
import React from 'react';
import renderer from'react-test-renderer';
import AddOrganization from './AddOrganization';
import config from '../../config';
import {MemoryRouter} from 'react-router-dom';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
const fetchMock = require('fetch-mock');

describe('AddOrganization Snapshot', () => {
    test('renders', () =>{
        const component = renderer.create(
            <MemoryRouter>
                <MuiThemeProvider>
                    <AddOrganization/>
                </MuiThemeProvider>
            </MemoryRouter>
        );
        let tree= component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});