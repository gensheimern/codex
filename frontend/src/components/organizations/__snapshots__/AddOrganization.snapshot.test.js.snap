// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddOrganization Snapshot renders 1`] = `
Array [
  <h4>
    Create Organization:
</h4>,
  <div
    className={undefined}
    style={
        Object {
            "backgroundColor": "transparent",
            "cursor": "auto",
            "display": "inline-block",
            "fontFamily": "Roboto, sans-serif",
            "fontSize": 16,
            "height": 72,
            "lineHeight": "24px",
            "muiPrepared": true,
            "position": "relative",
            "transition": "height 200ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
            "width": "100%",
          }
    }
>
    <label
        className={undefined}
        htmlFor="AOname"
        onClick={undefined}
        style={
            Object {
                "color": "rgba(0, 0, 0, 0.3)",
                "lineHeight": "22px",
                "muiPrepared": true,
                "pointerEvents": "none",
                "position": "absolute",
                "top": 38,
                "transform": "scale(1) translate(0, 0)",
                "transformOrigin": "left top",
                "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                "userSelect": "none",
                "zIndex": 1,
              }
        }
    >
        Name
    </label>
    <input
        disabled={false}
        id="AOname"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        style={
            Object {
                "WebkitOpacity": 1,
                "WebkitTapHighlightColor": "rgba(0,0,0,0)",
                "appearance": "textfield",
                "backgroundColor": "rgba(0,0,0,0)",
                "border": "none",
                "boxSizing": "border-box",
                "color": "rgba(0, 0, 0, 0.87)",
                "cursor": "inherit",
                "font": "inherit",
                "height": "100%",
                "marginTop": 14,
                "muiPrepared": true,
                "outline": "none",
                "padding": 0,
                "position": "relative",
                "width": "100%",
              }
        }
        type="text"
        value=""
    />
    <div>
        <hr
            aria-hidden="true"
            style={
                Object {
                    "borderBottomStyle": "solid",
                    "borderBottomWidth": 1,
                    "borderColor": "#e0e0e0",
                    "borderLeft": "none",
                    "borderRight": "none",
                    "borderTop": "none",
                    "bottom": 8,
                    "boxSizing": "content-box",
                    "margin": 0,
                    "muiPrepared": true,
                    "position": "absolute",
                    "width": "100%",
                  }
            }
        />
        <hr
            aria-hidden="true"
            style={
                Object {
                    "borderBottomStyle": "solid",
                    "borderBottomWidth": 2,
                    "borderColor": "#00bcd4",
                    "borderLeft": "none",
                    "borderRight": "none",
                    "borderTop": "none",
                    "bottom": 8,
                    "boxSizing": "content-box",
                    "margin": 0,
                    "muiPrepared": true,
                    "position": "absolute",
                    "transform": "scaleX(0)",
                    "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                    "width": "100%",
                  }
            }
        />
    </div>
</div>,
  <br />,
  <div
    className={undefined}
    style={
        Object {
            "backgroundColor": "transparent",
            "cursor": "auto",
            "display": "inline-block",
            "fontFamily": "Roboto, sans-serif",
            "fontSize": 16,
            "height": 72,
            "lineHeight": "24px",
            "muiPrepared": true,
            "position": "relative",
            "transition": "height 200ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
            "width": "100%",
          }
    }
>
    <label
        className={undefined}
        htmlFor="AOdescription"
        onClick={undefined}
        style={
            Object {
                "color": "rgba(0, 0, 0, 0.3)",
                "lineHeight": "22px",
                "muiPrepared": true,
                "pointerEvents": "none",
                "position": "absolute",
                "top": 38,
                "transform": "scale(1) translate(0, 0)",
                "transformOrigin": "left top",
                "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                "userSelect": "none",
                "zIndex": 1,
              }
        }
    >
        Description
    </label>
    <input
        disabled={false}
        id="AOdescription"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        style={
            Object {
                "WebkitOpacity": 1,
                "WebkitTapHighlightColor": "rgba(0,0,0,0)",
                "appearance": "textfield",
                "backgroundColor": "rgba(0,0,0,0)",
                "border": "none",
                "boxSizing": "border-box",
                "color": "rgba(0, 0, 0, 0.87)",
                "cursor": "inherit",
                "font": "inherit",
                "height": "100%",
                "marginTop": 14,
                "muiPrepared": true,
                "outline": "none",
                "padding": 0,
                "position": "relative",
                "width": "100%",
              }
        }
        type="text"
        value=""
    />
    <div>
        <hr
            aria-hidden="true"
            style={
                Object {
                    "borderBottomStyle": "solid",
                    "borderBottomWidth": 1,
                    "borderColor": "#e0e0e0",
                    "borderLeft": "none",
                    "borderRight": "none",
                    "borderTop": "none",
                    "bottom": 8,
                    "boxSizing": "content-box",
                    "margin": 0,
                    "muiPrepared": true,
                    "position": "absolute",
                    "width": "100%",
                  }
            }
        />
        <hr
            aria-hidden="true"
            style={
                Object {
                    "borderBottomStyle": "solid",
                    "borderBottomWidth": 2,
                    "borderColor": "#00bcd4",
                    "borderLeft": "none",
                    "borderRight": "none",
                    "borderTop": "none",
                    "bottom": 8,
                    "boxSizing": "content-box",
                    "margin": 0,
                    "muiPrepared": true,
                    "position": "absolute",
                    "transform": "scaleX(0)",
                    "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                    "width": "100%",
                  }
            }
        />
    </div>
</div>,
  <br />,
  <div
    className={undefined}
    style={
        Object {
            "backgroundColor": "transparent",
            "cursor": "auto",
            "display": "inline-block",
            "fontFamily": "Roboto, sans-serif",
            "fontSize": 16,
            "height": 72,
            "lineHeight": "24px",
            "muiPrepared": true,
            "position": "relative",
            "transition": "height 200ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
            "width": "100%",
          }
    }
>
    <label
        className={undefined}
        htmlFor="AOpassword"
        onClick={undefined}
        style={
            Object {
                "color": "rgba(0, 0, 0, 0.3)",
                "lineHeight": "22px",
                "muiPrepared": true,
                "pointerEvents": "none",
                "position": "absolute",
                "top": 38,
                "transform": "scale(1) translate(0, 0)",
                "transformOrigin": "left top",
                "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                "userSelect": "none",
                "zIndex": 1,
              }
        }
    >
        Password
    </label>
    <input
        disabled={false}
        id="AOpassword"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        style={
            Object {
                "WebkitOpacity": 1,
                "WebkitTapHighlightColor": "rgba(0,0,0,0)",
                "appearance": "textfield",
                "backgroundColor": "rgba(0,0,0,0)",
                "border": "none",
                "boxSizing": "border-box",
                "color": "rgba(0, 0, 0, 0.87)",
                "cursor": "inherit",
                "font": "inherit",
                "height": "100%",
                "marginTop": 14,
                "muiPrepared": true,
                "outline": "none",
                "padding": 0,
                "position": "relative",
                "width": "100%",
              }
        }
        type="password"
        value=""
    />
    <div>
        <hr
            aria-hidden="true"
            style={
                Object {
                    "borderBottomStyle": "solid",
                    "borderBottomWidth": 1,
                    "borderColor": "#e0e0e0",
                    "borderLeft": "none",
                    "borderRight": "none",
                    "borderTop": "none",
                    "bottom": 8,
                    "boxSizing": "content-box",
                    "margin": 0,
                    "muiPrepared": true,
                    "position": "absolute",
                    "width": "100%",
                  }
            }
        />
        <hr
            aria-hidden="true"
            style={
                Object {
                    "borderBottomStyle": "solid",
                    "borderBottomWidth": 2,
                    "borderColor": "#00bcd4",
                    "borderLeft": "none",
                    "borderRight": "none",
                    "borderTop": "none",
                    "bottom": 8,
                    "boxSizing": "content-box",
                    "margin": 0,
                    "muiPrepared": true,
                    "position": "absolute",
                    "transform": "scaleX(0)",
                    "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                    "width": "100%",
                  }
            }
        />
    </div>
</div>,
  <br />,
  <div
    className={undefined}
    style={
        Object {
            "backgroundColor": "transparent",
            "cursor": "auto",
            "display": "inline-block",
            "fontFamily": "Roboto, sans-serif",
            "fontSize": 16,
            "height": 72,
            "lineHeight": "24px",
            "muiPrepared": true,
            "position": "relative",
            "transition": "height 200ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
            "width": "100%",
          }
    }
>
    <label
        className={undefined}
        htmlFor="AOretypePassord"
        onClick={undefined}
        style={
            Object {
                "color": "rgba(0, 0, 0, 0.3)",
                "lineHeight": "22px",
                "muiPrepared": true,
                "pointerEvents": "none",
                "position": "absolute",
                "top": 38,
                "transform": "scale(1) translate(0, 0)",
                "transformOrigin": "left top",
                "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                "userSelect": "none",
                "zIndex": 1,
              }
        }
    >
        Retype password
    </label>
    <input
        disabled={false}
        id="AOretypePassord"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        style={
            Object {
                "WebkitOpacity": 1,
                "WebkitTapHighlightColor": "rgba(0,0,0,0)",
                "appearance": "textfield",
                "backgroundColor": "rgba(0,0,0,0)",
                "border": "none",
                "boxSizing": "border-box",
                "color": "rgba(0, 0, 0, 0.87)",
                "cursor": "inherit",
                "font": "inherit",
                "height": "100%",
                "marginTop": 14,
                "muiPrepared": true,
                "outline": "none",
                "padding": 0,
                "position": "relative",
                "width": "100%",
              }
        }
        type="password"
        value=""
    />
    <div>
        <hr
            aria-hidden="true"
            style={
                Object {
                    "borderBottomStyle": "solid",
                    "borderBottomWidth": 1,
                    "borderColor": "#e0e0e0",
                    "borderLeft": "none",
                    "borderRight": "none",
                    "borderTop": "none",
                    "bottom": 8,
                    "boxSizing": "content-box",
                    "margin": 0,
                    "muiPrepared": true,
                    "position": "absolute",
                    "width": "100%",
                  }
            }
        />
        <hr
            aria-hidden="true"
            style={
                Object {
                    "borderBottomStyle": "solid",
                    "borderBottomWidth": 2,
                    "borderColor": "#00bcd4",
                    "borderLeft": "none",
                    "borderRight": "none",
                    "borderTop": "none",
                    "bottom": 8,
                    "boxSizing": "content-box",
                    "margin": 0,
                    "muiPrepared": true,
                    "position": "absolute",
                    "transform": "scaleX(0)",
                    "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                    "width": "100%",
                  }
            }
        />
    </div>
</div>,
  <br />,
  <div
    className={undefined}
    style={
        Object {
            "WebkitTapHighlightColor": "rgba(0,0,0,0)",
            "backgroundColor": "#ffffff",
            "borderRadius": 2,
            "boxShadow": undefined,
            "boxSizing": "border-box",
            "color": "rgba(0, 0, 0, 0.87)",
            "display": "inline-block",
            "float": "right",
            "fontFamily": "Roboto, sans-serif",
            "minWidth": 88,
            "muiPrepared": true,
            "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          }
    }
>
    <button
        disabled={true}
        id="AOcreate"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        style={
            Object {
                "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
                "backgroundColor": "rgb(229, 229, 229)",
                "border": 10,
                "borderRadius": 2,
                "boxSizing": "border-box",
                "cursor": "default",
                "display": "inline-block",
                "fontFamily": "Roboto, sans-serif",
                "fontSize": "inherit",
                "fontWeight": "inherit",
                "height": 36,
                "lineHeight": "36px",
                "margin": 0,
                "muiPrepared": true,
                "outline": "none",
                "padding": 0,
                "position": "relative",
                "textAlign": "center",
                "textDecoration": "none",
                "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                "verticalAlign": null,
                "width": "100%",
              }
        }
        tabIndex={-1}
        type="button"
    >
        <div
            style={
                Object {
                    "backgroundColor": false,
                    "borderRadius": 2,
                    "height": 36,
                    "muiPrepared": true,
                    "top": 0,
                    "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                  }
            }
        >
            <span
                style={
                    Object {
                        "color": "rgba(0, 0, 0, 0.3)",
                        "fontSize": 14,
                        "fontWeight": 500,
                        "letterSpacing": 0,
                        "margin": 0,
                        "muiPrepared": true,
                        "opacity": 1,
                        "paddingLeft": 16,
                        "paddingRight": 16,
                        "position": "relative",
                        "textTransform": "uppercase",
                        "userSelect": "none",
                      }
                }
            >
                Create
            </span>
        </div>
    </button>
</div>,
]
`;
